### PR TITLE
docs: improve accessibility of attribute-binding example

### PR DESCRIPTION
**Pull request by kapunahelewong**
- Improves contrast.
- Adjusts colors and changes some styles so they aren't color dependent (to accommodate those with color blindness)
- Adds padding.
- Clarifies button copy.
- Removes unused styles.

## Before

Screen Shot 2021-04-02 at 3 45 07 PM

## After

Screen Shot 2021-04-02 at 3 21 40 PM

## Notes

I learned that 6% of the population is color blind (that means  6,000 out of every 100,000!). This is my first time  at accommodating for this particular concern, so if anyone else knows better than I do, please let me know so I can update the colors/styles accordingly. Some of my references included:

https://davidmathlogic.com/colorblind/#%23D81B60-%231E88E5-%23FFC107-%23004D40
https://www.color-hex.com/color-palette/49436
https://venngage.com/blog/color-blind-friendly-palette/

### CODE REVIEW

File: aio/content/examples/attribute-binding/src/app/app.component.css [hunk 1]
```diff
@@ -1,22 +1,32 @@
 .special {
-  background-color: #1976d2;
+  background-color: black;
```
Comment (AndrewKushnir) [inline aio/content/examples/attribute-binding/src/app/app.component.css:2]: We can probably use hex codes for consistency:
```suggestion
  background-color: #000000;
```

File: aio/content/examples/attribute-binding/src/app/app.component.css [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+h3 {
+  padding: 1rem 0 .75rem 0;
 }
 
-.new-class {
-  background-color: #ed1b2f;
-  font-style: italic;
-  color: #fff;
+.readability {
+  background-color: black;
```
Comment (AndrewKushnir) [inline aio/content/examples/attribute-binding/src/app/app.component.css:29]:
```suggestion
  background-color: #000000;
```

File: aio/content/examples/attribute-binding/src/app/app.component.html [hunk 3]
```diff
@@ -35,35 +35,41 @@ <h3>Basic specificity</h3>
...
+<!-- The `style.border` binding overrides any value for the `border` property in `styleExpression`.  -->
+<div [style.border]="border" [style]="styleExpression">Some text.</div>
 <!-- #enddocregion basic-specificity -->
 
 <!-- #docregion source-specificity -->
 <h3>Source specificity</h3>
 
 <!-- The `class.special` template binding overrides any host binding to the `special` class set by `dirWithClassBinding` or `comp-with-host-binding`.-->
-<comp-with-host-binding [class.special]="isSpecial" dirWithClassBinding>Some text.</comp-with-host-binding>
+<div>
```
Comment (AndrewKushnir) [inline aio/content/examples/attribute-binding/src/app/app.component.html:46]: Do we need extra `<div>`s here?

File: aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts [hunk 4]
```diff
@@ -2,18 +2,17 @@ import { Component, HostBinding } from '@angular/core';
 
 @Component({
   selector: 'comp-with-host-binding',
-  template: 'I am a component!',
+  template: `I am a component!`,
```
Comment (AndrewKushnir) [inline aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts:5]: I think we should revert this change as it's more common to use single quotes for strings without JS interpolations:
```suggestion
  template: 'I am a component!',
```

**Review by aikithoughts — APPROVED** [review]
Nothing blocking after @AndrewKushnir's feedback is addressed.

Reviewed-for: global-docs-approvers

File: aio/content/examples/attribute-binding/src/app/app.component.css (hunk 1 above)
Comment (gkalpak) [reply aio/content/examples/attribute-binding/src/app/app.component.css:2]: OOC, why this suggestion?
(Personally I don't mind as long as we are consistent, but @jelbourn [prefers using `black` instead of the HEX value][1] :wink:)

[1]: https://github.com/angular/angular/pull/40622#discussion_r567987909

File: aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts [hunk 5]
```diff
@@ -2,18 +2,17 @@ import { Component, HostBinding } from '@angular/core';
...
 })
 export class CompWithHostBindingComponent {
   @HostBinding('class.special')
   isSpecial = false;
 
   @HostBinding('style.color')
-  color = 'green';
+  color = '#F2A1D4';
```
Comment (gkalpak) [inline aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts:12]: Using a color name makes it easier for people to see the difference in the output.

File: aio/content/examples/attribute-binding/src/app/my-input-with-attribute-decorator.component.ts [hunk 6]
```diff
@@ -2,7 +2,7 @@ import { Attribute, Component } from '@angular/core';
 
 @Component({
   selector: 'app-my-input-with-attribute-decorator',
-  template: 'The type of the input is: {{ type }}'
+  template: `<p>The type of the input is: {{ type }}</p>`
```
Comment (gkalpak) [inline aio/content/examples/attribute-binding/src/app/my-input-with-attribute-decorator.component.ts:5]:
```suggestion
  template: '<p>The type of the input is: {{ type }}</p>'
```

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/content/examples/attribute-binding/src/app/app.component.css (hunk 1 above)
Comment (AndrewKushnir) [reply aio/content/examples/attribute-binding/src/app/app.component.css:2]: @gkalpak it just looks more consistent to use HEX, but I'm ok with other options if there is a strong preference :)

Comment (jelbourn) [reply aio/content/examples/attribute-binding/src/app/app.component.css:2]: My preference is to prefer the named color values when they are available as they are more semantically meaningful.

**Review by jelbourn — COMMENTED** [review]
+1 to other outstanding comments

File: aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts (hunk 5 above)
Comment (kapunahelewong) [reply aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts:12]: The only issue with this example is that the color green doesn't give enough contrast. Here it is with green:
Screen Shot 2021-04-29 at 1 35 59 PM

And here it is with #F2A1D4:

Screen Shot 2021-04-29 at 1 35 43 PM

The `comp-with-host-binding` component is used three times throughout the example to demonstrate different things. If we tweak the color and bg color too far for either one another becomes invisible. See the blue bar in the middle - that's actually this component:
Screen Shot 2021-04-29 at 1 46 13 PM

If we want to stick with accessible colors and styles that are easier to convey, we could instead create separate components. That would make this example a lot easier to style, imo. What do you think, @gkalpak?

File: aio/content/examples/attribute-binding/src/app/app.component.html [hunk 7]
```diff
@@ -35,35 +35,41 @@ <h3>Basic specificity</h3>
...
 <!-- #docregion dynamic-priority -->
 <h3>Dynamic vs static</h3>
 
 <!-- If `classExpression` has a value for the `special` class, this value overrides the `class="special"` below -->
 <div class="special" [class]="classExpression">Some text.</div>
 
-<!-- If `styleExpression` has a value for the `color` property, this value overrides the `style="color: blue"` below -->
-<div style="color: blue" [style]="styleExpression">Some text.</div>
+<!-- If `styleExpression` has a value for the `font-style` property, this value overrides the `style="font-style: bold"` below -->
+<div style="font-style: bold" [style]="styleExpression">Some text.</div>
```
Comment (gkalpak) [inline aio/content/examples/attribute-binding/src/app/app.component.html:62]: In order to showcase this, `styleExpression` must include a `font-style` style. [Currently it doesn't][1].

Maybe use `border` instead?

[1]: https://github.com/angular/angular/blob/dd55ded0ba5439af1b6452f6e0e7cf1ff21a1141/aio/content/examples/attribute-binding/src/app/app.component.ts#L13

File: aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts (hunk 5 above)
Comment (gkalpak) [reply aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts:12]: I am not too keen on creating many components, because that also increases the maintenance cost.
How about using a different property than the background color that does not interfere with the `color` (for example, using `border-color`)? 

##
BTW, I am not opposed to using a non-green color, but could we use one of the built-in color names?

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts (hunk 5 above)
Comment (kapunahelewong) [reply aio/content/examples/attribute-binding/src/app/comp-with-host-binding.component.ts:12]: Yes! I changed it to `pink`. I tried the border but it didn't look right and the text was then left out defaulting to `#444`. But fixed! Thank you!

File: aio/content/examples/attribute-binding/src/app/app.component.html (hunk 7 above)
Comment (kapunahelewong) [reply aio/content/examples/attribute-binding/src/app/app.component.html]: Changed to:

```
<!-- If `styleExpression` has a value for the `border` property, this value overrides 
the `style="border: dotted darkblue 3px"` below -->

<div style="border: dotted darkblue 3px" [style]="styleExpression">Some text.</div>
```

```
styleExpression = 'border: solid red 3px';
```

**Review by gkalpak — APPROVED** [review]
🚀
